Modules/Networking/Requests/DataUploadRequest.swift

```diff
@@ -73,3 +79,79 @@ public extension DataUploadRequest.Body {
     case exposureDetectionSummaries = "exposure_detection_summaries"
   }
 }
+
+// MARK: - Sorting
+
+extension DataUploadRequest.Body {
+  // In development, the key of the current day is also returned, resulting in a request with 15 TEKs. Given that the backend
+  // expects 14, the oldest one is discarded.
+  static func cap(_ teks: [CodableTemporaryExposureKey]) -> [CodableTemporaryExposureKey] {
+    let cappedTeks = teks
+      .sorted(by: CodableTemporaryExposureKey.byRollingStartNumberDesc)
+      .prefix(14)
+
+    return Array(cappedTeks)
+  }
+
+  static func cap(
+    _ summaries: [CodableExposureDetectionSummary],
+    maxSummaries: Int,
+    maxExposureInfo: Int
+  ) -> [CodableExposureDetectionSummary] {
+    let cappedSummaries = Array(
+      summaries
+        .sorted(by: CodableExposureDetectionSummary.byDateDescending)
```

Comment:
does ```.sorted(by: \.date)``` works?

Reply:
Nope, `sorted` expects a `(Type, Type) -> Bool` closure. We had a `sorted(picking: (Type) -> Void)` in the extensions at some point, but then we removed it.
Honestly I feel this is readable enough.